### DISCUSSION TITLE
Pydantic overriding default value with Field produces Pyright error, is it expected?

### DISCUSSION THREAD

**Discussion by rafrafek** [microsoft/pyright#8226] [Q&A] [answered]
Hello,

I'm getting this error:

```
"expire_time" overrides a field of the same name but is missing a default value
```

for this code:

```python
from datetime import timedelta

from pydantic import BaseModel, Field


class BaseSession(BaseModel):
    expire_time: timedelta = timedelta(hours=3)


class CustomSession(BaseSession):
    expire_time: timedelta = Field(timedelta(hours=3), alias="expireTime")
    # ^^^^^^^^^ missing a default value
```

Is it expected, or is it a bug in Pydantic or Pyright? Isn't Field a default value here?

**Comment by erictraut** [comment, marked as answer]
This is expected behavior. It's neither a bug in pydantic nor in pyright.

Pydantic provides a non-standard way to specify a default value via a positional argument, as you've done in your code sample. Field specifiers in stdlib `dataclass` do not support this; they require you to specify the default by keyword argument `default=`.

Standards-compliant type checkers like pyright do not know about pydantics idiosyncrasies. Their behavior is based on the [dataclass_transform](https://typing.readthedocs.io/en/latest/spec/dataclasses.html#the-dataclass-transform-decorator) mechanism which assumes standard dataclass behaviors. 

Pydantic also supports the standard dataclass behavior, so if you change your code to the following, it will satisfy both pydantic and standards-based type checkers like pyright.

```python
class CustomSession(BaseSession):
    expire_time: timedelta = Field(default=timedelta(hours=3), alias="expireTime")
```

**Reply by rafrafek** [reply]
Thank you for clarifying it for me!
